### PR TITLE
UCP/TEST: Fix setting active message callbacks for ASYNC transports.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/959/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1566/ for details.

**Comment by yosefe** [comment]
@MattBBaker @alex-mikheev @shamisp

File: src/uct/base/uct_iface.c [hunk 1]
```diff
@@ -72,8 +73,8 @@ ucs_status_t uct_iface_set_am_handler(uct_iface_h tl_iface, uint8_t id,
         return UCS_OK;
     }
 
-    if (flags == 0) {
-        ucs_debug("am cb flags must not be empty");
+    if (!(flags & (UCT_AM_CB_FLAG_SYNC|UCT_AM_CB_FLAG_ASYNC))) {
+        ucs_debug("invalid active message flags 0x%x", flags);
         return UCS_ERR_INVALID_PARAM;
```
Comment (shamisp) [inline src/uct/base/uct_iface.c]: Maybe it should be converted to an error.

**Comment by MattBBaker** [comment]
@yosefe :+1:

File: src/uct/base/uct_iface.c (hunk 1 above)
Comment (shamisp) [inline src/uct/base/uct_iface.c]: @yosefe  - please address the comment

Comment (yosefe) [inline src/uct/base/uct_iface.c]: will fix in next PR

Comment (shamisp) [inline src/uct/base/uct_iface.c]: @yosefe - really appreciate this

**Pull request merged**